### ISSUE TITLE
fix: double submit with Enter

### ISSUE THREAD

**Issue by iotalambda** [microsoft/fluentui-blazor#1050]
# 🐛 Bug Report

When navigating to an EditForm's submit button by keyboard and hitting Enter, submit occurs twice.

## 💻 Repro or Code Sample

`dotnet new install Microsoft.FluentUI.AspNetCore.Templates::4.1.1`
`dotnet new fluentblazor`

Replace Counter.razor content with the following, tab your way to the FluentButton and hit Enter -> current count is incremented by 2.
```razor
@page "/counter"
@rendermode InteractiveServer

<PageTitle>Counter</PageTitle>

<h1>Counter</h1>

<p role="status">Current count: <FluentBadge Appearance="Appearance.Neutral">@currentCount</FluentBadge></p>

<EditForm OnValidSubmit=@IncrementCount Model="new { }">
    <button type="submit">Submit with normal button -> +1</button>
    <FluentButton Appearance="Appearance.Accent" Type=@ButtonType.Submit>Submit with FluentButton -> +2</FluentButton>
</EditForm>

@code {
    private int currentCount = 0;

    private void IncrementCount()
    {
        currentCount++;
    }
}
```

## 🌍 Your Environment

Occurs at least in Edge and Firefox

**Comment by vnbaaij** [comment]
I just tried it with same template and your exact code. Not seeing this behavior:
![counter-#1050](https://github.com/microsoft/fluentui-blazor/assets/1761079/b98180da-396d-4f52-be0a-75fe72db5332)

**Comment by iotalambda** [comment]
> I just tried it with same template and your exact code. Not seeing this behavior: ![counter-#1050](https://private-user-images.githubusercontent.com/1761079/287291086-b98180da-396d-4f52-be0a-75fe72db5332.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE0NDE5NjYsIm5iZiI6MTcwMTQ0MTY2NiwicGF0aCI6Ii8xNzYxMDc5LzI4NzI5MTA4Ni1iOTgxODBkYS0zOTZkLTRmNTItYmUwYS03NWZlNzJkYjUzMzIuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMDFUMTQ0MTA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzFmMjNjMTQ0MDk2Y2UwNWE0MDQxMzlmZGU3YjM3MTljY2MyNjcxOTE5M2Q3ZmYzNzk0ZGYwNzA0NmI1ZGM3ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.sSZQgr55Sus7wPYc9X6kWYtUBo_fQD5-ocBhD98cDdM) [ ![counter-#1050](https://private-user-images.githubusercontent.com/1761079/287291086-b98180da-396d-4f52-be0a-75fe72db5332.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE0NDE5NjYsIm5iZiI6MTcwMTQ0MTY2NiwicGF0aCI6Ii8xNzYxMDc5LzI4NzI5MTA4Ni1iOTgxODBkYS0zOTZkLTRmNTItYmUwYS03NWZlNzJkYjUzMzIuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMDFUMTQ0MTA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzFmMjNjMTQ0MDk2Y2UwNWE0MDQxMzlmZGU3YjM3MTljY2MyNjcxOTE5M2Q3ZmYzNzk0ZGYwNzA0NmI1ZGM3ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.sSZQgr55Sus7wPYc9X6kWYtUBo_fQD5-ocBhD98cDdM) ](https://private-user-images.githubusercontent.com/1761079/287291086-b98180da-396d-4f52-be0a-75fe72db5332.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE0NDE5NjYsIm5iZiI6MTcwMTQ0MTY2NiwicGF0aCI6Ii8xNzYxMDc5LzI4NzI5MTA4Ni1iOTgxODBkYS0zOTZkLTRmNTItYmUwYS03NWZlNzJkYjUzMzIuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMDFUMTQ0MTA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzFmMjNjMTQ0MDk2Y2UwNWE0MDQxMzlmZGU3YjM3MTljY2MyNjcxOTE5M2Q3ZmYzNzk0ZGYwNzA0NmI1ZGM3ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.sSZQgr55Sus7wPYc9X6kWYtUBo_fQD5-ocBhD98cDdM) [ ](https://private-user-images.githubusercontent.com/1761079/287291086-b98180da-396d-4f52-be0a-75fe72db5332.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE0NDE5NjYsIm5iZiI6MTcwMTQ0MTY2NiwicGF0aCI6Ii8xNzYxMDc5LzI4NzI5MTA4Ni1iOTgxODBkYS0zOTZkLTRmNTItYmUwYS03NWZlNzJkYjUzMzIuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMDFUMTQ0MTA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzFmMjNjMTQ0MDk2Y2UwNWE0MDQxMzlmZGU3YjM3MTljY2MyNjcxOTE5M2Q3ZmYzNzk0ZGYwNzA0NmI1ZGM3ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.sSZQgr55Sus7wPYc9X6kWYtUBo_fQD5-ocBhD98cDdM)

Did you submit by clicking the button? The bug does not occur when submitting by clicking - only when submitting by activating the button and then hitting Enter on your keyboard.

**Comment by vnbaaij** [comment]
Ok, can reproduce with `Enter` keypress...weird.
While we search for a solution, a workaround is to remove the `Type=ButtonType.Submit` part from the code. The `IncrementCount` method will then only be called once.

**Comment by vnbaaij** [comment]
Of course, removing the submit is not alway going to be a viable solution. 

I've traced back the bug all the way to the FAST source code (which is what is used to build the `fluent-button` web component that we wrap in our `FluentButton` component): https://github.com/microsoft/fast/issues/6704.

Unfortunately, it needs to be fixed there first. After that we need to get a new release of the web components script and integrate that in our library. I'm in contact with the FAST maintainers to see if and how this ca nbe expidited but,  I'm afraid, until then this behavior can not be changed/circumvented when usen the keyboard.

**Comment by iotalambda** [comment]
Thanks @vnbaaij. Too bad the problem lies so deep.

Also, another workaround is to use a non-fluent submit button for the time being.

**Comment by vnbaaij** [comment]
No, nothing has changed. We anticipate it wont be fixed until we have the new web components

**Comment by niltor** [comment]
I encountered a similar problem.
When implementing an edit form,
I used FluentTextArea and needed to enter multiple lines of text.
When the text content has a value (from table row), pressing Enter will directly submit instead of wrapping in the textarea!

However, if the textarea itself has no value, you can enter and press Enter normally, which is very strange!

```razor
<EditForm OnValidSubmit="@SaveAsync" EditContext="editContext">
    <DataAnnotationsValidator />
    <FluentValidationSummary />

    <FluentDialogBody>
        <FluentStack Orientation="Orientation.Vertical">
            <FluentTextField @bind-Value="AddDto!.ClientName"
                             Label="@Localizer.Get(LanguageKey.ClientName)"
                             Placeholder="@Localizer.Get(LanguageKey.ClientName)"
                             class="w-100" Required />
            <FluentValidationMessage For="() => AddDto.ClientName"></FluentValidationMessage>

            <FluentTextArea Rows="3" @bind-Value="RedirectUris"
                            @onkeydown:stopPropagation
                            Placeholder="@Localizer.Get(LanguageKey.RedirectUris)" Class="w-100"
                            Label="@Localizer.Get(LanguageKey.RedirectUris)"
                            Required />
            <FluentLabel Typo="Typography.Body" Color="Color.Info">
                @Localizer.Get(LanguageKey.RowValueTip)
            </FluentLabel>
            <FluentValidationMessage For="() => AddDto.RedirectUris"></FluentValidationMessage>

        </FluentStack>
    </FluentDialogBody>

    <FluentDialogFooter>
        <FluentButton Appearance="Appearance.Accent" Type="ButtonType.Submit">
            @Localizer.Get(LanguageKey.Save)
        </FluentButton>
        <FluentButton Appearance="Appearance.Neutral" OnClick="@CancelAsync">
            @Localizer.Get(LanguageKey.Cancel)
        </FluentButton>
    </FluentDialogFooter>
</EditForm>
```

![Image](https://github.com/user-attachments/assets/091d06e1-6ea0-446a-9204-1057c89bb121)

**Comment by nodefusioncom** [comment]
I had the same issue, but the work around was to set the disabled property inside submit button, FluentButton,  and bind the Disabled property to bool IsSubmitting. Inside the method first set IsSubmitting variable to true and call StateHasChanged method.
The second action will not be called because the button is disabled.
````

<FluentEditForm Model="@myModel" OnValidSubmit="OnSubmit" Context="myFormContext">
    <FluentButton Appearance="Appearance.Outline" Type="ButtonType.Submit" Disabled="@IsSubmitting">
        Save
    </FluentButton>
</FluentEditForm>
...
private bool IsSubmitting = false;
private async Task OnSubmit()
{
   IsSubmitting  = true;
   StateHasChanged(); 
........
   IsSubmitting  = false;
   StateHasChanged(); 

}

````

**Comment by nfn-100004** [comment]
I had the same issue, but the work around was to set the disabled property inside submit button, FluentButton,  and bind the Disabled property to bool IsSubmitting. Inside the method first set IsSubmitting variable to true and call StateHasChanged method.
The second action will not be called because the button is disabled.
````

<FluentEditForm Model="@myModel" OnValidSubmit="OnSubmit" Context="myFormContext">
    <FluentButton Appearance="Appearance.Outline" Type="ButtonType.Submit" Disabled="@IsSubmitting">
        Save
    </FluentButton>
</FluentEditForm>
...
private bool IsSubmitting = false;
private async Task OnSubmit()
{
   IsSubmitting  = true;
   StateHasChanged(); 
........
   IsSubmitting  = false;
   StateHasChanged(); 

}

````

**Comment by MarvinKlein1508** [comment]
This cannot be fixed in v4. In the current v5 branch I've tested your provided code and it's working fine there. For this reason I'm closing this issue.
